### ISSUE TITLE
Implement a focus management workaround for Mobile Safari anchor link focus bug

### ISSUE THREAD

**Issue by cfarm** [wagtail/wagtail#5459]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Mobile Safari on iOS 11 and up has introduced a bug on anchor links. When a user navigates a page using Voiceover, clicking anchor links moves the viewport to the linked section of the page but does not move the focus to that section. Users are unable to navigate pages using anchor links because of this.

There is a blog post documenting this bug, with suggested JavaScript workarounds here: https://axesslab.com/skip-links/

I would recommend implementing one of the JS workarounds since this significantly degrades the user experience for Voiceover users, especially once we have implemented a skip nav link for #5406. This bug would render the skip nav link and other anchors essentially useless for affected users.

### Steps to Reproduce

1. On an iPhone running iOS 11 or later, open Safari and navigate to any website with an anchor link https://webaim.org or https://www.consumerfinance.gov (I don't know of any examples in the Wagtail admin, since we don't have a skip nav link yet)
2. Turn on Voiceover
3. Swipe right to navigate through focusable links
4. Double tap to click the "Skip to main content" link
5. Notice the viewport has moved down the page
6. Swipe right again, if you are sighted you'll see the focus has not moved to the new section of the page and if you are listening to the Voiceover screen reader you will hear focusable content that came after the skip link instead of the focusable content on the anchor section of the page.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

> * I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: 

No, not at this time, this will become a bug once #5406 is implemented

**Comment by thibaudcolas** [comment]
Related #5406.
